### PR TITLE
blockchain: Switch to FindSpentTicketsInBlock.

### PR DESCRIPTION
**Pull request by matheusd**
First try at improving #913.

This adds a new function `FindSpentTicketsInBlock` that extracts all spent and revoked tickets, plus the vote bits of a given block and uses that function instead of the individual functions when loading a new node into the blockchain.

This improves startup time a bit, as the `isSSGen`, `isSSRtx`, and `determineTxType` are somewhat slow, as they need to decode the output script of the transactions. Looping over the transactions once and doing a single test is faster.

The first commit adds a couple of instrumentation lines (to be removed before final merge).

Numbers for my computer:
```
| variation     | mainnet | testnet |
| with patch    |   23.7s |   42.6s |
| without patch |   35.9s |  118.0s |
```

### CODE REVIEW

File: blockchain/chain.go [hunk 1]
```diff
@@ -625,8 +625,9 @@ func (b *BlockChain) loadBlockNode(dbTx database.Tx, hash *chainhash.Hash) (*blo
 	}
 
 	blockHeader := block.MsgBlock().Header
-	node := newBlockNode(&blockHeader, ticketsSpentInBlock(block),
-		ticketsRevokedInBlock(block), voteBitsInBlock(block))
+	tickets := ticketsInBlock(block)
+	node := newBlockNode(&blockHeader, tickets.spentTickets, tickets.revokedTickets,
```
Comment (dajohi) [inline blockchain/chain.go]: i wonder newBlockNode should just take a ticketsInBlockInfo now...

File: blockmanager.go [hunk 2]
```diff
@@ -2614,12 +2614,20 @@ func newBlockManager(s *server, indexManager blockchain.IndexManager) (*blockMan
 		return nil, err
 	}
 
+	// FIXME remove before merge
+	tstart := time.Now().UnixNano()
```
Comment (dajohi) [inline blockmanager.go]: remove

File: blockmanager.go [hunk 3]
```diff
@@ -2614,12 +2614,20 @@ func newBlockManager(s *server, indexManager blockchain.IndexManager) (*blockMan
...
+
 	// Retrieve the current previous block hash and next stake difficulty.
 	curPrevHash := bm.chain.BestPrevHash()
 	nextStakeDiff, err := bm.chain.CalcNextRequiredStakeDifficulty()
 	if err != nil {
 		return nil, err
 	}
+	bmgrLog.Infof("Next required Stake difficulty: %d", nextStakeDiff)
+
+	// FIXME remove before merge
```
Comment (dajohi) [inline blockmanager.go]: remove

File: blockchain/chain.go (hunk 1 above)
Comment (matheusd) [reply blockchain/chain.go]: Could be helpful, yeah.

Do you wanna wait for someone else to chime in, or should I go ahead and do it?

**Pull request merged**